quantum/util.h

```diff
@@ -16,6 +16,28 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 #pragma once
 
+/* GCC ≥ 10 and Clang have __has_builtin() & __has_attribute(); until then… */
+
+#ifndef __has_builtin
+#    define __has_builtin(_) 0
+#endif
+
+#ifndef __has_attribute
+#    define __has_attribute(_) 0
+#endif
+
+#if defined(__GNUC__) || __has_attribute(noinline)
+#    define Q_NEVER_INLINE __attribute__((noinline))
```

Comment:
LUFA already defines these two as `ATTR_NO_INLINE` and `ATTR_ALWAYS_INLINE`:
https://github.com/qmk/lufa/blob/HEAD/LUFA/Common/Attributes.h#L91

We should use those names and fall back to these defines when not compiling a LUFA board.